shared/js/icons_helper.js

```diff
@@ -48,7 +48,16 @@
     iconTargetSize = iconTargetSize * window.devicePixelRatio;
 
     // First look for an icon in the manifest.
-    if (siteObj.webManifestUrl && siteObj.webManifest) {
+    if (siteObj.manifest) {
```

Comment:
Given that I only know of one app which has both a Firefox App manifest and a W3C manifest at the moment, it isn't urgent, but what you did is fine, thank you :)